Source/spinal-harmony.cs

```diff
@@ -1,314 +1,327 @@
+            {
+                Log.Message("SOS2 spinal engines: spinalcomp in SpinalRecalc null");
+                return null;
+            }
+            if (instance.parent.Map == null)
+            {
+                Log.Message("SOS2 spinal engines: parent map in SpinalRecalc was null!");
+                return null;
+            }
+            vecMoveCheck = -1 * buildingPointer.Rotation.FacingCell; // OG SOS2 basically rebuilt FacingCell, someone pointed that out to me
+            //Log.Message(string.Format("found rotation in SpincalRecalc, iterating, vec is: {0}", vecMoveCheck));
+            previousThingPos = buildingPointer.Position - (2 * vecMoveCheck); //Engines are inverse side from turrets, and engines are lorge
+            while (!foundNonAmp)
             {
-                CompSpinalMount spinalComp = instance.parent.TryGetComp<CompSpinalMount>();
-                if (spinalComp == null)
+                previousThingPos -= vecMoveCheck;
+                buildingPointer = previousThingPos.GetFirstThingWithComp<CompSpinalEngineMount>(instance.parent.Map);
+                if (buildingPointer == null)
                 {
-                    Log.Message("spinalcomp in SpinalRecalc null");
-                    return 0;
+                    //Log.Message($"SOS2 spinal engines:  new amp at {previousThingPos} in SpinalRecalc was null!");
+                    foundNonAmp = true; //Currently unused in this case but might need it for a later plan
+                    break;
                 }
-                if (instance.parent.Map == null)
+                CompSpinalEngineMount ampComp = buildingPointer.TryGetComp<CompSpinalEngineMount>();
+                //Log.Message($"SOS2 Spinal Engines: vecs are: pointer, {buildingPointer.Position} Previous {previousThingPos}");
+                if (buildingPointer.Rotation.Opposite != instance.parent.Rotation) // Remember, engines vs amps are inverted
                 {
-                    Log.Message("parent map in SpinalRecalc was null!");
-                    return 0;
+                    //Log.Message("SOS2 spinal engines: amps rotation in SpinalRecalc did not match parent rotation");
+                    toReturn.playerError = "Backwards spinal component found!";
+                    return toReturn;
                 }
-                vecMoveCheck = -1 * amp.Rotation.FacingCell; // OG SOS2 basically rebuilt FacingCell, someone pointed that out to me
-                // Log.Message(string.Format("found rotation in SpincalRecalc, iterating, vec is: {0}", vec));
-                previousThingPos = amp.Position - (2 * vecMoveCheck); //Engines are inverse side from turrets, and engines are lorge
-                while (!foundNonAmp)
+                //This is... A way, to check I guess. (buildingPointer = prevthingppos.getfirstcomp<spinal> above returns thing, if that is *also* the center, 1-wide, must? be amp) 
+                if (buildingPointer.Position == previousThingPos)
```

Comment:
Right I think you had previously mentioned this is kind of confusing. A problem with "I had to immerse myself in it so it makes sense in my head". Fundamentally, due to the fact that the buildingPointer isn't null (would have bailed), it found *something* one more tile away while iterating over the stack, that has a CompSpinalEngineMount, and also isn't reversed. (The one tile thing is because of the -= vecMoveCheck). So, it's a single tile wide, thus it must be an engine accelerator (ampComp is perhaps a poor name here because it used to find amplifiers, but now accelerators?). The buildingPointer is the building, but the ampComp is a "component" that is attached to the building that stores data, does logic, etc, so they aren't the same. The ampComp we found means the building has comp indicating it's a part of a spinal engine, but not what part. This is using the size to evaluate that because otherwise I'd have to check... The defname directly? Which feels brittle if I ever update names etc. Would this perhaps be less ugly to read if I just updated my ompProperties_SpinalEngineMount with a... bool? For if something is an accelerator? hrm